rules/subrepo_rules.build_defs

```diff
@@ -151,6 +160,7 @@ def new_local_repository(name:str, path:str):
 
 def github_repo(name:str, repo:str, revision:str, build_file:str=None, hashes:str|list=None,
                 strip_prefix:str=None, strip_build:bool=False, config:str=None,
+                access_token:str=None,
```

Comment:
This looks like it's the actual access token as a string. We can't have people checking that into their codebase. We should pass this in via a secret. Doesn't github support basic auth? We could pass through the username and password file fields.

Reply:
I was going to pass the string $GITHUB_TOKEN to use the env var present in GitHub Actions and in Codespaces. I'm not against secret files. I just wasn't sure how that would work. Would that require an absolute path? Or is the idea to make it relative to step up out of the git project? 

Re: basic auth on GitHub: https://developer.github.com/changes/2020-02-14-deprecating-password-auth/

Reply:
Thoughts on this @Tatskaari? I'm happy to help fix how you want it. I'm starting to ramp up other teams and need to start sharing build defs in private repos.